### DISCUSSION TITLE
Why is ButtonInteraction#constructor private?

### DISCUSSION THREAD

**Discussion by AdaptingAFM** [discordjs/discord.js#7870] [Q&A] [answered]
Hey, I'm domain typing my application and it so happens that whenever an interaction of type button and a given id occurs, the message acted upon will always have the same structure. 

I consider the embeds and their many different components a domain view that I present to my users: they are structured in classes that I serialize to at the controller level to easily operate. These days I have been trying to type my ButtonInteractions, and I cannot as the constructor is private and I need a runtime reference for my DI. Are there any specific reasons why this is an exception over the other type of interactions? The SlashCommands' one is public as an example.

**Comment by manuelvleeuwen** [comment, marked as answer]
You shouldn't instantiate a ButtonInteraction yourself, that's why the constructor is private.

**Reply by AdaptingAFM** [reply]
> You shouldn't instantiate a ButtonInteraction yourself, that's why the constructor is private.

None of the interactions are meant to be instantiated by a user yet not all the constructors are private, and I will not instantiate it: I will use it as a runtime reference.

**Reply by monbrey** [reply]
Thanks for pointing this out - we should make the others private too

**Reply by AdaptingAFM** [reply]
@monbrey 
What's the point in restricting the access to constructors, if them who were to misuse the structure would only experience non-exceptional unexpected behavior? By privatizing them you block advanced users of extending classes for reflection, and I am literally forced to use a ts transformer to reflect at runtime on structures

**Reply by almostSouji** [reply]
what exactly prevents you from using functions and standalone maps (if you need to augment additional data) and passing the structure to it?
that'd be the advanced, clean, way to augment functionality
